crates/cxx-qt-lib/include/assertion_utils.h

```diff
@@ -0,0 +1,16 @@
+// clang-format off
+// SPDX-FileCopyrightText: 2022 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// clang-format on
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+// SPDX-FileContributor: Leon Matthes <leon.matthes@kdab.com>
+// SPDX-FileContributor: Matt Aber <matt.aber@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+#pragma once
+
+#define assert_alignment_and_size(TYPE, EXP_ALIGN, MEMBERS)                    \
```

Comment:
guess there is also the question, do we need to pass in the expected alignment? What are we trying to compare here? Can we compare the struct alignment to the Qt type? Or do we always just need to check that it is `alignof(::std::size_t)` ?

Reply:
true we can just do this now that we define a type
```cpp
static_assert(alignof(s##TYPE) == alignof(TYPE));
static_assert(sizeof(s##TYPE) == sizeof(TYPE))
```